### PR TITLE
Add mass fraction diffusion coefficients

### PR DESCRIPTION
**Pull request by totto82**
Depends on https://github.com/OPM/opm-common/pull/3878

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this opm-simulators=5143 please

File: opm/models/blackoil/blackoildiffusionmodule.hh [hunk 1]
```diff
@@ -61,6 +61,13 @@ class BlackOilDiffusionModule<TypeTag, /*enableDiffusion=*/false>
     using RateVector = GetPropType<TypeTag, Properties::RateVector>;
 
 public:
+
+    /*!
+     * \brief Initialize all internal data structures needed by the brine module
+     */
+    static void initFromState(const EclipseState&)
```
Comment (atgeirr) [inline opm/models/blackoil/blackoildiffusionmodule.hh:68]: Unlike its sibling, this trivial version is not guarded by HAVE_ECL_INPUT, I guess it should be?

File: opm/models/blackoil/blackoildiffusionmodule.hh [hunk 2]
```diff
@@ -196,22 +215,28 @@ public:
     }
 
 private:
-    static Scalar toMolFractionGasOil (unsigned regionIdx) {
-        Scalar mMOil = FluidSystem::molarMass(FluidSystem::oilCompIdx, regionIdx);
+    static Scalar toFractionGasOil (unsigned regionIdx) {
```
Comment (atgeirr) [inline opm/models/blackoil/blackoildiffusionmodule.hh:218]: I think this is correct. However, I wish for some comments documenting the mathematical model, and in particular explaining this function.

**Review by atgeirr — COMMENTED** [review]
Looks ok, must probably add some HAVE_ECL_INPUT guards.

File: opm/models/blackoil/blackoildiffusionmodule.hh (hunk 1 above)
Comment (totto82) [reply opm/models/blackoil/blackoildiffusionmodule.hh:69]: Added

File: opm/models/blackoil/blackoildiffusionmodule.hh (hunk 2 above)
Comment (totto82) [reply opm/models/blackoil/blackoildiffusionmodule.hh:239]: I added a description like for the dispersion model.

**Comment by totto82** [comment]
jenkins build this opm-simulators=5143 please

**Pull request merged**